File: lib/tools/display-format.ts
// Formatage d'affichage — notation courte de bookmaker (1/X/2, Over/Under,
// BTTS) reconnue par les parieurs habitués à 1xBet et équivalents, plutôt
// que les phrases longues utilisées en interne pour le calcul des résultats
// (lib/tools/result-checker.ts continue d'utiliser bet_type tel quel).

export function shortenBetType(betType: string): string {
  const bt = betType.toLowerCase()

  const overMatch = bt.match(/(?:plus de|over)\s*(\d+(?:\.\d+)?)/)
  if (overMatch) return `Over ${overMatch[1]}`

  const underMatch = bt.match(/(?:moins de|under)\s*(\d+(?:\.\d+)?)/)
  if (underMatch) return `Under ${underMatch[1]}`

  if (bt.includes('btts') || bt.includes('deux équipes marquent')) {
    return bt.includes('non') ? 'BTTS Non' : 'BTTS Oui'
  }

  if (bt.includes('victoire') && (bt.includes('domicile') || bt.includes('home'))) return '1'
  if (bt.includes('victoire') && (bt.includes('extérieur') || bt.includes('exterieur') || bt.includes('away'))) return '2'
  if (bt.includes('match nul') || bt === 'nul' || bt.includes('draw')) return 'X'

  const handicapMatch = betType.match(/handicap\s+(.+?)\s+([+-]?\d+(?:\.\d+)?)/i)
  if (handicapMatch) {
    const [, team, point] = handicapMatch
    return `${team.trim()} ${point.startsWith('-') || point.startsWith('+') ? point : `+${point}`}`
  }

  return betType
}

/** Initiales pour le badge d'équipe (pas de vrai logo — voir note dans image-generator.ts). */
export function teamInitials(name: string): string {
  const words = name.trim().split(/\s+/).filter(Boolean)
  if (words.length >= 2) return (words[0][0] + words[1][0]).toUpperCase()
  return name.slice(0, 2).toUpperCase()
}

File: backend/tools/display_format.py
"""Port de lib/tools/display-format.ts — notation courte de bookmaker."""
import re


def shorten_bet_type(bet_type: str) -> str:
    bt = bet_type.lower()

    over_match = re.search(r"(?:plus de|over)\s*(\d+(?:\.\d+)?)", bt)
    if over_match:
        return f"Over {over_match.group(1)}"

    under_match = re.search(r"(?:moins de|under)\s*(\d+(?:\.\d+)?)", bt)
    if under_match:
        return f"Under {under_match.group(1)}"

    if "btts" in bt or "deux équipes marquent" in bt:
        return "BTTS Non" if "non" in bt else "BTTS Oui"

    if "victoire" in bt and ("domicile" in bt or "home" in bt):
        return "1"
    if "victoire" in bt and ("extérieur" in bt or "exterieur" in bt or "away" in bt):
        return "2"
    if "match nul" in bt or bt == "nul" or "draw" in bt:
        return "X"

    handicap_match = re.search(r"handicap\s+(.+?)\s+([+-]?\d+(?:\.\d+)?)", bet_type, re.IGNORECASE)
    if handicap_match:
        team, point = handicap_match.group(1), handicap_match.group(2)
        sign = point if point.startswith(("-", "+")) else f"+{point}"
        return f"{team.strip()} {sign}"

    return bet_type


def team_initials(name: str) -> str:
    """Initiales pour le badge d'équipe (pas de vrai logo)."""
    words = [w for w in name.strip().split() if w]
    if len(words) >= 2:
        return (words[0][0] + words[1][0]).upper()
    return name[:2].upper()
